sled-storage/src/manager.rs

```diff
@@ -815,6 +992,131 @@ impl StorageManager {
         }
     }
 
+    // Lists datasets that this zpool contains
+    async fn datasets_list(
+        &self,
+        zpool: &ZpoolName,
+    ) -> Result<Vec<DatasetProperties>, Error> {
+        let log = self.log.new(o!("request" => "datasets_list"));
+
+        let datasets_of_interest = [
+            // We care about the zpool itself, and all direct children.
+            zpool.to_string(),
+            // Likewise, we care about the encrypted dataset, and all
+            // direct children.
+            format!("{zpool}/{CRYPT_DATASET}"),
+            // The zone dataset gives us additional context on "what zones
+            // have datasets provisioned".
+            format!("{zpool}/{ZONE_DATASET}"),
+        ];
+
+        info!(log, "Listing datasets within zpool"; "zpool" => zpool.to_string());
+        illumos_utils::zfs::Zfs::get_dataset_properties(
+            datasets_of_interest.as_slice(),
+        )
+        .map_err(Error::Other)
+    }
```

Comment:
It might be nice to document the motivation for doing this within the `StorageManager` task --- i'm guessing performing this in the actor helps to make sure that datasets won't be created/destroyed whilst we are attempting to list them?

Reply:
so, full admission, I added it here to de-duplicate a call that wanted to access the same information - it just seemed "under the purview of storage management", so I put it here. But yes! Serializing the request would ensure that datasets aren't concurrently being created / destroyed concurrently with this operation.

Added some docs